src/mat/impls/inv/matinv.c

```diff
@@ -172,53 +172,21 @@ static PetscErrorCode MatInvComputeNullSpace_Inv(Mat imat)
   }
   TRY( MatAssemblyBegin(R,MAT_FINAL_ASSEMBLY) );
   TRY( MatAssemblyEnd(R,MAT_FINAL_ASSEMBLY) );
+  //TODO mem leak?
+  TRY( MatOrthColumns(R, orthType, orthForm, &R, NULL) );
```

Comment:
I do not think that orthogonalization should be part of MatComputeNullSpaceMat().

Reply:
* I believe the nullspace base being ortonormal is a usual requirement. `MatNullSpace` [assumes this](https://www.mcs.anl.gov/petsc/petsc-current/docs/manualpages/Mat/MatNullSpaceCreate.html).
* If one doesn't want to perform the orthonormalization,`MAT_ORTH_NONE` can be passed.
* This line taken from `QPTDualize` was actually wrong and would require an additional variable and additional `MatDestroy` since `MatOrthColumns` has nothing like a `MatReuse` argument.

Reply:
OK, it makes sense to choose orthType.